test/v1alpha1/service.go

```diff
@@ -346,8 +347,16 @@ func IsServiceReady(s *v1alpha1.Service) (bool, error) {
 	return s.Generation == s.Status.ObservedGeneration && s.Status.IsReady(), nil
 }
 
-// IsServiceNotReady will check the status conditions of the service and return true if the service is
-// not ready.
+// IsServiceNotReady will check the status conditions of the service and return false if the service is
+// ready.
 func IsServiceNotReady(s *v1alpha1.Service) (bool, error) {
-	return s.Generation == s.Status.ObservedGeneration && !s.Status.IsReady(), nil
+	result := s.Status.GetCondition(v1alpha1.ServiceConditionReady)
+	return result != nil && result.Status == corev1.ConditionFalse, nil
+}
+
+// IsServiceNotRouteReady checks the RouteReady status of the service and return false if the service's
+// RouteReady is true.
+func IsServiceNotRouteReady(s *v1alpha1.Service) (bool, error) {
```

Comment:
nit: The condition is actually "RoutesReady" it makes more sense grammatically if the "not' is moved.
```suggestion
func IsServiceRoutesNotReady(s *v1alpha1.Service) (bool, error) {
```